pkg/beacon/analyzer.go

```diff
@@ -323,31 +327,47 @@ func getTsHistogramScore(min int64, max int64, tsList []int64) ([]int64, []int,
 		cv = 1.0
 	}
 
-	// calculate first potential score
-	// histograms with bigger flat sections will score higher, up to 4 flat sections
-	// this will score well for graphs that have flat sections with a big distance between them,
-	// i.e. a beacon that alternates between 1 and 5 connections per hour
-	// This score will only be calculated if the number of total bars on the histogram fills at
-	// least half the day (>11), otherwise this could lead to a false positive and not fill the
+	cvScore := math.Ceil((1.0-float64(cv))*1000) / 1000
+	if cvScore > 1.0 {
+		cvScore = 1.0
+	}
+
+	// Calculate second potential score
+	// this will score well for graphs that have 2-3 flat sections in their connection histogram,
+	// or a bimodal freqCount histogram.
+	// Example - a beacon that alternates between 1 and 5 connections per hour
+	// This score will only be calculated if the number of total bars on the histogram is at
+	// least 10, otherwise this could lead to a false positive and not fill the
 	// original intent of this scoring.
-	score1 := 0.0
+	bimodalFit := float64(0)
+
+	if totalBars > 10 {
+		largest := 0
+		secondLargest := 0
+
+		// get top two frequency mode bars
+		for key, value := range freqCount {
+			if key > 0 {
```

Comment:
Good catch!

I was trying to avoid more if statements in the for loop that makes the freq count map. I mistakenly moved the check here. I'll get that fixed.